lib/ui/recommendation.js

```diff
@@ -154,53 +159,92 @@ Recommendation.prototype = {
     this.events.publish('recommendation-mousemove', evt);
   },
   show: function(data) {
-    if (this.timeout) {
-      this.win.clearTimeout(this.timeout);
-      this.timeout = null;
+    if (this.hideTimeout) {
+      this.win.console.log('show clearing hideTimeout');
+      this.win.clearTimeout(this.hideTimeout);
+      this.hideTimeout = null;
+    }
+
+    if (!this.el) {
+      return;
     }
 
-    // Don't re-render the recommendation if the content hasn't changed since
-    // the previous recommendation.
-    if (this.prev && this.el && this.el.firstChild && data.result.url === this.prev.result.url) {
+    this.win.console.log('about to dupe check inside show().');
+
+    // Reuse the recommendation if it's still in the DOM and the content hasn't
+    // changed since the previous recommendation.
+    if (this.el.firstChild && data && this.row &&
+        this.row.url === data.url) {
       this.el.collapsed = false;
```

Comment:
What about using the idea from the description for #241 here: instead of just unsetting `this.el.collapsed`, you should remove and reseat it in the XUL DOM to trigger a re-layout: `rec.parentNode.replaceChild(rec, rec)`?